aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -195,10 +199,16 @@ Tensor& absolute_(Tensor& self) {
 }
 
 Tensor& angle_out(Tensor& result, const Tensor& self) {
-  return unary_op_impl_with_complex_to_float_out(result, self, angle_stub);
+  return unary_op_impl_with_complex_to_float_out(result, self, angle_stub, /*promotes_integer_to_float=*/true);
 }
 Tensor angle(const Tensor& self) {
-  return unary_op_impl_with_complex_to_float(self, at::angle_out);
+  if (self.is_complex()) {
+    const auto float_type = c10::toValueType(self.scalar_type());
```

Comment:
Won't this run afoul of the safe casting logic in TensorIterator?

Reply:
Oh, sorry, it doesn't because it doesn't call tensor iterator, it calls angle_out.

Reply:
This is actually similar to what was already there.

https://github.com/pytorch/pytorch/blob/f5b68e74d75c38d5e2044fc6b62112181080bb3f/aten/src/ATen/native/UnaryOps.cpp#L105-L109